nav2_tasks/include/nav2_tasks/is_stuck_condition.hpp

```diff
@@ -137,35 +124,12 @@ class IsStuckCondition : public BT::ConditionNode, public rclcpp::Node
     while (spinning_ok_) {
```

Comment:
Doesn't really need a separate node with its own thread. If you received the Node::SharedPtr, you could use this to create any pubs/subs. The responsibility to spin the node is with the caller/user of this class. In general, I'd say that simpler is better if the simpler version implements the same functionality. In this case, you can have fewer threads and not have to manage the separate thread.

Reply:
Agree, with the simplifications I introduced the separate thread can be removed. I will test it out.

Reply:
@mjeronimo As discussed offline, this will require a few changes, and as you suggested even to the BT library. Since this is not critical, I'll push as is and followup on #469 later.